### PR TITLE
[E2E] Causal Analysis - Aggregate Causal Analysis - diabetes-decision-making notebook - checking tests failing in pipeline only

### PR DESCRIPTION
**Pull request by vinuthakaranth**
[E2E] Causal Analysis - Aggregate Causal Analysis - diabetes-decision-making notebook

## Description

Adding tests for Aggregate Causal analysis for notebook https://github.com/microsoft/responsible-ai-toolbox/blob/main/notebooks/responsibleaidashboard/responsibleaidashboard-diabetes-decision-making.ipynb

## Areas changed

npm packages changed:

- [ ] responsibleai/causality
- [ ] responsibleai/core-ui
- [ ] responsibleai/counterfactuals
- [ ] responsibleai/dataset-explorer
- [ ] responsibleai/fairness
- [ ] responsibleai/interpret
- [ ] responsibleai/localization
- [ ] responsibleai/mlchartlib
- [ ] responsibleai/model-assessment

Python packages changed:

- [ ] raiwidgets
- [ ] responsibleai
- [ ] erroranalysis
- [ ] rai_core_flask

## Tests

- [ ] No new tests required.
- [x] New tests for the added feature are part of this PR.
- [ ] I validated the changes manually.

## Documentation:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/widget-e2e/src/describer/modelAssessment/causalAnalysis/describeAggregateCausalAffects.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        cy.get(`${Locators.CausalChartXAxisValues} text`)
+          .last()
+          .invoke("text")
+          .then((text) => {
+            expect(
+              dataShape.causalAnalysisData?.featureListInCausalTable
+            ).include(text);
+          });
+      });
+      it("should have continuos and binary treatment definitions", () => {
```
Comment (imatiach-msft) [inline apps/widget-e2e/src/describer/modelAssessment/causalAnalysis/describeAggregateCausalAffects.ts:77]: nit minor spelling correction: continuos -> continuous

File: apps/widget-e2e/src/util/ScatterChartErrorBars.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  public sortByH(): IScatter[] {
+    return this.Elements.sort((a, b) => a.top - b.top);
+  }
+  public clickNthPoint(idx: number): void {
+    const offset = this.getNthPointOffset(idx);
+    if (!offset) {
+      return;
+    }
+    cy.get(`${this.container} .nsewdrag.drag`).trigger("mouseover", {
```
Comment (imatiach-msft) [inline apps/widget-e2e/src/util/ScatterChartErrorBars.ts:31]: wow, this is hovering on points in the scatter plot to trigger mouseover events? very fancy test code

File: apps/widget-e2e/src/util/ScatterChartErrorBars.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { Chart, IChartElement } from "./Chart";
+
+const dReg =
```
Comment (imatiach-msft) [inline apps/widget-e2e/src/util/ScatterChartErrorBars.ts:6]: interesting, just for my understanding, how does this regular expression work?

**Review by imatiach-msft — APPROVED** [review]

File: apps/widget-e2e/src/describer/modelAssessment/causalAnalysis/describeAggregateCausalAffects.ts (hunk 1 above)
Comment (vinuthakaranth) [reply apps/widget-e2e/src/describer/modelAssessment/causalAnalysis/describeAggregateCausalAffects.ts:77]: Thank you, good catch!

File: apps/widget-e2e/src/util/ScatterChartErrorBars.ts (hunk 2 above)
Comment (vinuthakaranth) [reply apps/widget-e2e/src/util/ScatterChartErrorBars.ts:31]: yeas, mouseover along with trigger created hover effect. But I am yet to find the correct offset in this case. It is not completely producing hover result yet. I will have tweak a little bit to provide hover result.

File: apps/widget-e2e/src/util/ScatterChartErrorBars.ts (hunk 3 above)
Comment (vinuthakaranth) [reply apps/widget-e2e/src/util/ScatterChartErrorBars.ts:6]: d attribute is something like below:
 M1182.6,272.53h8m-4,0V342.5m-4,0h8.
This format is from svg grammar ([link](https://www.w3.org/TR/SVG/paths.html)) for drawing graphs. We extract numbers from here and use x/y/heights etc to verify the graph is generated as expected.

**Pull request merged**